tm/src/main/java/io/seata/tm/api/DefaultFailureHandlerImpl.java

```diff
@@ -36,10 +58,54 @@ public void onBeginFailure(GlobalTransaction tx, Throwable cause) {
     @Override
     public void onCommitFailure(GlobalTransaction tx, Throwable cause) {
         LOGGER.warn("Failed to commit transaction[" + tx.getXid() + "]", cause);
+        timer.newTimeout(new CheckTimerTask(tx, GlobalStatus.Committed), SCHEDULE_INTERVAL_SECONDS, TimeUnit.SECONDS);
     }
 
     @Override
     public void onRollbackFailure(GlobalTransaction tx, Throwable cause) {
         LOGGER.warn("Failed to rollback transaction[" + tx.getXid() + "]", cause);
+        timer.newTimeout(new CheckTimerTask(tx, GlobalStatus.Rollbacked), SCHEDULE_INTERVAL_SECONDS, TimeUnit.SECONDS);
+    }
+
+    protected class CheckTimerTask implements TimerTask{
+
+        private final GlobalTransaction tx;
+
+        private final GlobalStatus required;
+
+        private int count = 0;
+
+        private boolean isStopped = false;
+
+        protected CheckTimerTask(final GlobalTransaction tx, GlobalStatus required) {
+            this.tx = tx;
+            this.required = required;
+        }
+
+        @Override
+        public void run(Timeout timeout) throws Exception {
+            if(!isStopped){
+                if(++count > RETRY_MAX_TIMES){
+                    LOGGER.error("transaction[" + tx.getXid() + "] retry fetch status times exceed the limit [" + RETRY_MAX_TIMES + " times]");
+                    return;
+                }
+                isStopped = shouldStop(tx, required);
```

Comment:
the globalTransaction maybe should be update from server?

Reply:
I don't understand what you mean...